nomad/container.py

```diff
@@ -165,18 +167,28 @@ def is_stopped(self):
         return self._container.status_code == constants.CONTAINER_STOPPED
 
     @property
+    def lxd_name(self):
+        """ Returns the name of the container that is used in the scope of LXD.
+
+        This name id supposed to be unique among all the containers managed by LXD.
+        """
+        # Note: all container names must be a valid hostname! That is: maximum 63 characters, no
+        # dots, no digit at first position, made entirely of letters/digits/hyphens, ...
+        if not hasattr(self, '_lxd_name'):
+            lxd_name_prefix = '{project_name}-{name}'.format(
```

Comment:
I'll come back with a new commit but I don't see why we should use `lxd_name` in log messages (5) - we are using "local names" with the nomad command (which seems fine IMO).

My proposal: we could assign an auto-generated name if needed as you said (2) - eg. "default" - if we are considering a single-container project. If the project contains more than one container, we could enforce the definition of container names in the configuration file... or we could just enforce container names in all cases.